packages/swapper/src/swappers/ButterSwap/xhr.ts

```diff
@@ -0,0 +1,210 @@
+import { fromAssetId, solanaChainId } from '@shapeshiftoss/caip'
+import type { Asset } from '@shapeshiftoss/types'
+import { chainIdToFeeAssetId } from '@shapeshiftoss/utils'
+import type { Result } from '@sniptt/monads'
+import { Err, Ok } from '@sniptt/monads'
+import { zeroAddress } from 'viem'
+
+import type { SwapErrorRight } from '../../types'
+import { TradeQuoteError } from '../../types'
+import { makeSwapErrorRight } from '../../utils'
+import type {
+  BridgeInfo,
+  BridgeInfoApiResponse,
+  BuildTxResponse,
+  FindTokenResponse,
+  RouteResponse,
+  SupportedChainListResponse,
+} from './types'
+import { butterHistoryService } from './utils/butterSwapHistoryService'
+import { butterService } from './utils/butterSwapService'
+import { chainIdToButterSwapChainId } from './utils/helpers'
+
+export enum ButterSwapErrorCode {
+  ParameterError = 2000,
+  ChainNotSupported = 2001,
+  TokenNotSupported = 2002,
+  NoRouteFound = 2003,
+  InsufficientLiquidity = 2004,
+  SlippageOutOfRange = 2005,
+  InsufficientAmount = 2006,
+  InvalidAddress = 2007,
+}
+
+type ButterSwapPromise<T> = Promise<Result<T, SwapErrorRight>>
+
+/**
+ * @see https://docs.butternetwork.io/butter-swap-integration/butter-api-for-routing/get-supportedchainlist
+ */
+export const getSupportedChainList = async (): ButterSwapPromise<SupportedChainListResponse> => {
+  const result = await butterService.get<SupportedChainListResponse>('/supportedChainList')
+  if (result.isErr()) return Err(result.unwrapErr())
+  // No runtime validation, just return the data
+  return Ok(result.unwrap().data)
+}
+
+/**
+ * @see https://docs.butternetwork.io/butter-swap-integration/butter-api-for-routing/get-findtoken
+ */
+export const findToken = async (
+  chainId: number,
+  address: string,
+): ButterSwapPromise<FindTokenResponse> => {
+  const result = await butterService.get<FindTokenResponse>('/findToken', {
+    params: { chainId, address },
+  })
+  if (result.isErr()) return Err(result.unwrapErr())
+  return Ok(result.unwrap().data)
+}
+
+/**
+ * @see https://docs.butternetwork.io/butter-swap-integration/butter-api-for-routing/get-route
+ */
+export interface GetButterRouteArgs {
+  sellAsset: Asset
+  buyAsset: Asset
+  amountHumanUnits: string
```

Comment:
for consistency, and also to ensure that naming reflects this is a sell amount (some swappers, which we don't support in the app, do allow reverse quotes by expected buy amount) 

```suggestion
  sellAmountCryptoBaseUnit: string
```